src/Component.js

```diff
@@ -1,73 +1,51 @@
 /* Represents a component of a vector */
-class Component
-{
-	constructor(length, maxRep, jump, hop, promote)
-	{
+class Component {
+	constructor(length, maxRep, jump, hop, promote) {
 		this.length = length;
 		/* Slight efficiency, clean output while debugging */
 		this.maxRep = length !== 0 ? maxRep : 1;
 		this.jump = jump;
 		this.hop = hop;
 		this.promote = promote;
 	}
-	
-	static Create(componentString, globalString)
-	{
-		const comp = new Component(0, 1, false, false, false);
-		comp.length = Number(componentString.match(/(-?\d+)/g)[0]);
-		
-		/* Check for jump on first component or end */
-		if (componentString.includes("j") || globalString.includes("j"))
-		{
-			comp.jump = true;
-		}
-		
-		/* Check for hop on first component or end */
-		if (componentString.includes("h") || globalString.includes("h"))
-		{
-			comp.hop = true;
-		}
-		
-		/* Check for promote on first component or end */
-		if (componentString.includes("p") || globalString.includes("p"))
-		{
-			comp.promote = true;
-		}
-		
-		/* Check for finite repetition on end */
-		const finiteEndRepetition = globalString.match(/{(\d+)}/g);
-		if (finiteEndRepetition)
-		{
-			comp.maxRep = Number(finiteEndRepetition[0].slice(2, -1));
-		}
-		
-		/* Check for infinite repetition */
-		if(componentString.includes("+") || globalString.includes("+"))
-		{
-			comp.maxRep = 100; /* arbitrarily large */
-		}
 
-		/* Check for finite specific repeition */
-		const finiteSpecificRepetition = componentString.match(/{(\d+)}/);
-		if (finiteSpecificRepetition)
-		{
-			comp.maxRep = Number(finiteSpecificRepetition[1]);
+	static Create(localFlags, globalFlags) {
+		const component = new Component(
+			Number(localFlags.match(/(-?\d+)/g)[0]),
+			1,
+			localFlags.includes("j") || globalFlags.includes("j"),
+			localFlags.includes("h") || globalFlags.includes("h"),
+			localFlags.includes("p") || globalFlags.includes("p")
+		);
+
+		const globalFiniteRepetition = globalFlags.match(/{(\d+)}/);
+		const globalInfiniteRepetition = globalFlags.includes("+");
+		const localFiniteRepetition = localFlags.match(/{(\d+)}/);
+		const localInfiniteRepetition = localFlags.includes("+");
+
+		// Check for global finite and infinite repetition
+		if (globalFiniteRepetition && globalInfiniteRepetition) {
+			throw "ERROR: Component global flags contain both a finite and an infinite repetition flag";
```

Comment:
Illegal repetition is checked in Vector as well. We should probably only check here, since this will catch all cases, but the Vector check will not catch when a Component is directly created.